### PR TITLE
Disallow credentials overwriting during registration

### LINKED ISSUE #55: Disallow overwriting credentials
PIN-protected Credential can't be removed without PIN, but can be overwritten by registering new one with the same name.
This inconsistency might be confusing for the user. Either disallow overwriting credentials with the same name, or separate logically PIN- and HW-based KEKs.

### PR DESCRIPTION
**Pull request by szszszsz**
Disallow credentials overwriting during registration
Require explicit removal with Delete() command from now on.

Fixes #55

### CODE REVIEW

File: src/state.rs [hunk 1]
```diff
@@ -179,6 +180,10 @@ impl State {
         (Err(encrypted_container::Error::FailedDecryption), None)
     }
 
+    pub fn file_exists<T: FilesystemClient>(&mut self, trussed: &mut T, filename: PathBuf) -> bool {
+        try_syscall!(trussed.read_file(self.location, filename)).is_ok()
```
Comment (sosthene-nitrokey) [inline src/state.rs:184]: The [`entry_metadata`](https://github.com/trussed-dev/trussed/blob/main/src/client.rs#L649C12-L649C12) syscall was made for this use case.

**Review by sosthene-nitrokey — CHANGES_REQUESTED** [review]

**Comment by sosthene-nitrokey** [comment]
Seeing #55 I wonder why the decision to not allow deletion of PIN protected credentials without the PIN was made.

**Comment by szszszsz** [comment]
> Seeing #55 I wonder why the decision to not allow deletion of PIN protected credentials without the PIN was made.

In other words, you see advantageous to add PIN-protected credential removal operation, which would not require the PIN. Can you elaborate on that?

From my perspective, the PIN-protected credential operations should always require PIN, including removal. The intention here was to protect more from the accidental breaking of the workflow (UX), than from a DOS attack.

File: src/state.rs (hunk 1 above)
Comment (szszszsz) [reply src/state.rs:184]: Thanks! Will change to it. 
Note that the name is not really intuitive for this use case though. Does it not call for a proper alias?

Comment (sosthene-nitrokey) [reply src/state.rs:184]: You mean check_exists?
Entry metadata does more than just checking existence.

**Comment by sosthene-nitrokey** [comment]
I just think that from a security standpoint deletion without PIN is not an issue (you could anyway do it with factory-reset), so I don't see why it would not be allowed. We'll see if someone asks for this feature I suppose

**Comment by szszszsz** [comment]
> I just think that from a security standpoint deletion without PIN is not an issue (you could anyway do it with factory-reset), so I don't see why it would not be allowed. We'll see if someone asks for this feature I suppose

Can you generate new keys on OpenPGP card without PIN?

File: src/state.rs (hunk 1 above)
Comment (szszszsz) [reply src/state.rs:184]: `check_exists` is not a good name. There is `exists` in the API already, and it refers to a key, not a file.
`file_exists` is the intuitive one as far as I see.

Comment (szszszsz) [reply src/state.rs]: Moved check to entry_metadata. Resolving.

**Review by sosthene-nitrokey — APPROVED** [review]

**Pull request merged**